x/staketia/keeper/migration.go

```diff
@@ -0,0 +1,159 @@
+package keeper
+
+import (
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	recordtypes "github.com/Stride-Labs/stride/v22/x/records/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v22/x/stakeibc/types"
+	oldtypes "github.com/Stride-Labs/stride/v22/x/staketia/legacytypes"
+	"github.com/Stride-Labs/stride/v22/x/staketia/types"
+)
+
+// TODO [UPGRADE HANDLER]: Migrate stakeibc host zone (set redemptions enabled to true on each host zone)
+
+// Helper to deserialize the host zone with the old types
+func (k Keeper) GetLegacyHostZone(ctx sdk.Context) (hostZone oldtypes.HostZone, err error) {
+	store := ctx.KVStore(k.storeKey)
+	hostZoneBz := store.Get(types.HostZoneKey)
+
+	if len(hostZoneBz) == 0 {
+		return hostZone, types.ErrHostZoneNotFound.Wrapf("No HostZone found, there must be exactly one HostZone!")
+	}
+
+	k.cdc.MustUnmarshal(hostZoneBz, &hostZone)
+	return hostZone, nil
+}
+
+// Update the newly created stakeibc host zone with the accounting values from staketia
+func (k Keeper) UpdateStakeibcHostZone(ctx sdk.Context, legacyHostZone oldtypes.HostZone) (stakeibctypes.HostZone, error) {
+	// Grab the newly created stakeibc host zone
+	stakeibcHostZone, found := k.stakeibcKeeper.GetHostZone(ctx, types.CelestiaChainId)
+	if !found {
+		return stakeibctypes.HostZone{}, errors.New("celestia host zone not found in stakeibc after registration")
+	}
+
+	// Disable redemptions and set the redemption rate to the one from stakeibc
+	stakeibcHostZone.RedemptionsEnabled = false
+	stakeibcHostZone.RedemptionRate = legacyHostZone.LastRedemptionRate
```

Comment:
Why `LastRedemptionRate` instead of `RedemptionRate`?

Reply:
It should be `RedemptionRate`. It must have been [fixed](https://github.com/Stride-Labs/stride/pull/1223/files#diff-e29b0f57bc104470ceb763d46fdec16efae60e2708e7e98957c6eb060bd82f21R45-R49) at some point after this PR